dot/peerset/peerset.go

```diff
@@ -729,3 +769,29 @@ func (ps *PeerSet) doWork(ctx context.Context) {
 		}
 	}
 }
+
+func (ps *PeerSet) periodicallyAllocateSlots(ctx context.Context) {
+	ticker := time.NewTicker(ps.nextPeriodicAllocSlots)
+
+	defer func() {
+		ticker.Stop()
+		close(ps.resultMsgCh)
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil {
+				logger.Debugf("peerset context finished with error: %w", err)
```

Comment:
a context always finishes with an error :wink: 

Also don't use `%w` for logging errors, but `%s`

```suggestion
				logger.Debugf("peerset slot allocation exiting: %s", err)
```